utility/environment_check.py

```diff
@@ -0,0 +1,127 @@
+"""
+Util for environment check befora and after test to compare and find stale
+leftovers
+"""
+import logging
+import pytest
+
+from ocs import ocp, defaults, constants, exceptions
+from deepdiff import DeepDiff
+
+log = logging.getLogger(__name__)
+
+
+POD = ocp.OCP(kind=constants.POD)
+SC = ocp.OCP(kind=constants.STORAGECLASS)
+CEPHFILESYSTEM = ocp.OCP(kind=constants.CEPHFILESYSTEM)
+CEPHBLOCKPOOL = ocp.OCP(kind=constants.CEPHBLOCKPOOL)
+PV = ocp.OCP(kind=constants.PV)
+PVC = ocp.OCP(kind=constants.PVC)
+SECRET = ocp.OCP(kind=constants.SECRET)
+NS = ocp.OCP(kind=constants.NAMESPACE)
+
+ADDED_RESOURCE = 'iterable_item_added'
+REMOVED_RESOURCE = 'iterable_item_removed'
+
+
+@pytest.fixture(scope='class')
+def environment_checker(request):
+    request.addfinalizer(get_status_after_execution)
+    get_status_before_execution()
+
+
+def get_status_before_execution():
+    """
+
+    """
+    defaults.ENV_STATUS_PRE['pod'] = POD.get(all_namespaces=True)['items']
+    defaults.ENV_STATUS_PRE['sc'] = SC.get(all_namespaces=True)['items']
+    defaults.ENV_STATUS_PRE['cephfs'] = CEPHFILESYSTEM.get(
+        all_namespaces=True
+    )['items']
+    defaults.ENV_STATUS_PRE['cephbp'] = CEPHBLOCKPOOL.get(
+        all_namespaces=True
+    )['items']
+    defaults.ENV_STATUS_PRE['pv'] = PV.get(all_namespaces=True)['items']
+    defaults.ENV_STATUS_PRE['pvc'] = PVC.get(all_namespaces=True)['items']
+    defaults.ENV_STATUS_PRE['secret'] = SECRET.get(
+        all_namespaces=True
+    )['items']
+    defaults.ENV_STATUS_PRE['namespace'] = NS.get(all_namespaces=True)['items']
+
+
+def get_status_after_execution():
+    """
+
+    """
+    defaults.ENV_STATUS_POST['pod'] = POD.get(all_namespaces=True)['items']
+    defaults.ENV_STATUS_POST['sc'] = SC.get(all_namespaces=True)['items']
+    defaults.ENV_STATUS_POST['cephfs'] = CEPHFILESYSTEM.get(
+        all_namespaces=True
+    )['items']
+    defaults.ENV_STATUS_POST['cephbp'] = CEPHBLOCKPOOL.get(
+        all_namespaces=True
+    )['items']
+    defaults.ENV_STATUS_POST['pv'] = PV.get(all_namespaces=True)['items']
+    defaults.ENV_STATUS_POST['pvc'] = PVC.get(all_namespaces=True)['items']
+    defaults.ENV_STATUS_POST['secret'] = SECRET.get(
+        all_namespaces=True
+    )['items']
+    defaults.ENV_STATUS_POST['namespace'] = NS.get(
+        all_namespaces=True
+    )['items']
```

Comment:
can you please elaborate on what is the issue?
Not sure I understand this "get all pods for given list of kinds"

Reply:
Method OCP.get() is using inner state of OCP object to get kind and namespace. If there were a function which takes both as an argument instead, we could do something like:

```
kind_list = ["pod", "sc", "cephfs", ... ]
env_status = {}
for kind in kind_list:
    env_status[kind] = ocp_get(kind=kind, all_namespacess=True)
```

without initializing 8 different objects for each kind. It would also make it possible to move the code above into function which could be just called with env_status_pre in setup, and env_status_post in teardown.

Reply:
creating such a function ocp_get() is pretty easy but I really want this checks to be part of the infrastructure as soon as possible cause there are many existing test cases that not cleaning the env after finish - this will enforce it.

As soon as this will be merged, I will implement such function and change the calls

Reply:
creating such a function ocp_get() is pretty easy but I really want this checks to be part of the infrastructure as soon as possible cause there are many existing test cases that not cleaning the env after finish - this will enforce it.

As soon as this will be merged, I will implement such function and change the calls

Reply:
Agreed, as I noted in the [1st comment here](https://github.com/red-hat-storage/ocs-ci/pull/237/files#r294251297), we don't need to address it in this PR.

Reply:
@mbukatov  It can be just a util function, dont know what it has to do with OCP class design.

Reply:
https://github.com/red-hat-storage/ocs-ci/issues/248